garden/src/features/gardens/components/create/CreateGardenFormFields.tsx

```diff
@@ -211,8 +211,11 @@ export const CreateGardenFormFields = () => {
       </div>
 
       <div className="mt-8 flex justify-end gap-2">
-        <Button type="submit" className={"inline-block"}>
-          {isSubmitting ? "Creating Garden..." : `Create ${isTestGarden ? "Test": ""} Garden`}
+        <Button type="submit" className={"inline-block"} disabled={
+          isSubmitting || (searchParams.get("type") === "modal" &&
+            (!!form.formState.errors.modal || !form.getValues("modal.file_contents")))
```

Comment:
I'm surprised that we need logic specific to the `file_contents` field here. Is this something we can encode in `garden.types.ts` instead?